transport/resolver.go

```diff
@@ -0,0 +1,23 @@
+// Copyright 2019 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package transport
```

Comment:
*discussion* Is a DNS resolver a `transport`? Maybe it would be better to put it in `network`? @fortuna any thoughts?

Reply:
I'd prefer to put this in a separate `dns` package at the root, since we usually have to deal with DNS separately, and it makes it clearer we have utilities for DNS.